### PR TITLE
feat(builtins): rego files diagnostics with opa check

### LINKED ISSUE #924: Add support for linting rego files via opa cli
### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues) and there are no existing ones with the same request.

### Feature description

It will be useful to have support for `*.rego` files for [OPA engine](https://www.openpolicyagent.org/) that can be linted with `opa check` command.

### Help

Yes

### Implementation help

_No response_

### PR DESCRIPTION
**Pull request by eshepelyuk**
Resolves #924

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/opacheck.lua [hunk 1]
```diff
@@ -0,0 +1,49 @@
+local h = require("null-ls.helpers")
+local methods = require("null-ls.methods")
+
+local handle_opacheck_output = function(params)
+    local diags = {}
+    if params.output.errors == nil then
+        return diags
+    end
+    for _, d in ipairs(params.output.errors) do
+        if d.location ~= nil then
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/opacheck.lua:10]: I'm curious - what do diagnostics without locations look like? Are they still worth showing? If we leave the location empty, they'll show up at the top of the file (which may or may not be useful).

File: lua/null-ls/builtins/diagnostics/opacheck.lua [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        url = "https://www.openpolicyagent.org/docs/latest/cli/#opa-check",
+        description = "Check Rego source files for parse and compilation errors.",
+    },
+    method = methods.internal.DIAGNOSTICS_ON_SAVE,
+    filetypes = { "rego" },
+    generator_opts = {
+        command = "opa",
+        args = { "check", "-f", "json", "--strict", "$ROOT/src/", "$ROOT/test/" },
+        format = "json",
+        to_stdin = false,
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/opacheck.lua:37]: No need to explicitly specify these `false` options, so we can take them out.

**Review by jose-elias-alvarez — COMMENTED** [review]
Just a couple of clarifying questions but this looks good, thank you!

File: lua/null-ls/builtins/diagnostics/opacheck.lua (hunk 2 above)
Comment (eshepelyuk) [reply lua/null-ls/builtins/diagnostics/opacheck.lua:37]: Should i remove all the declarations with false value ?

File: lua/null-ls/builtins/diagnostics/opacheck.lua (hunk 1 above)
Comment (eshepelyuk) [reply lua/null-ls/builtins/diagnostics/opacheck.lua:10]: i only noticed one case - when incorrect parameters are passed to opa check.

File: lua/null-ls/builtins/diagnostics/opacheck.lua (hunk 2 above)
Comment (jose-elias-alvarez) [reply lua/null-ls/builtins/diagnostics/opacheck.lua:37]: Yep, not a major issue but I think that would be the cleanest.

Comment (eshepelyuk) [reply lua/null-ls/builtins/diagnostics/opacheck.lua:37]: ok, will do

Comment (eshepelyuk) [reply lua/null-ls/builtins/diagnostics/opacheck.lua]: @jose-elias-alvarez done

**Comment by jose-elias-alvarez** [comment]
Thanks!

**Pull request merged**